rewrite-xml/src/main/java/org/openrewrite/xml/ChangeNamespaceValue.java

```diff
@@ -0,0 +1,141 @@
+    private static final String VERSION_PREFIX = "version";
+
+    @Override
+    public String getDisplayName() {
+        return "Change XML Attribute of a specific resource version";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Alters XML Attribute value within specified element of a specific resource versions.";
+    }
+
+    @Nullable
+    @Option(displayName = "Element name",
+            description = "The name of the element whose attribute's value is to be changed. Interpreted as an XPath Expression.",
+            example = "property",
+            required = false)
+    String elementName;
+
+    @Nullable
+    @Option(displayName = "Old value",
+            description = "Only change the property value if it matches the configured `oldValue`.",
+            example = "newfoo.bar.attribute.value.string",
+            required = false)
+    String oldValue;
+
+    @Option(displayName = "New value",
+            description = "The new value to be used for the namespace.",
+            example = "newfoo.bar.attribute.value.string")
+    String newValue;
+
+    @Nullable
+    @Option(displayName = "Resource version",
+            description = "The version of resource to change",
+            example = "1.1",
+            required = false)
+    String versionMatcher;
+
+    @Option(displayName = "Search All Namespaces",
+            description = "Specify whether evaluate all namespaces",
+            example = "true")
+    boolean searchAllNamespaces;
+
+
+    @JsonCreator
+    public ChangeNamespaceValue(@Nullable @JsonProperty("elementName") String elementName, @Nullable @JsonProperty("oldValue") String oldValue,
+                                @NonNull @JsonProperty("newValue") String newValue, @Nullable @JsonProperty("versionMatcher") String versionMatcher,
+                                @NonNull @JsonProperty("searchAllNamespaces") boolean searchAllNamespaces) {
+        this.elementName = elementName;
+        this.oldValue = oldValue;
+        this.newValue = newValue;
+        this.versionMatcher = versionMatcher;
+        this.searchAllNamespaces = searchAllNamespaces;
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new XmlIsoVisitor<ExecutionContext>() {
+
+            String version = null;
+
+            @Override
+            public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag t = super.visitTag(tag, ctx);
+                this.version = null;
+                if (elementName == null || new XPathMatcher(elementName).matches(getCursor())) {
+                    // find versions
+                    t = t.withAttributes(ListUtils.map(t.getAttributes(), this::visitChosenElementAttribute));
+                }
+
+                if (this.version != null) {
+                    //change namespace
+                    t = t.withAttributes(ListUtils.map(t.getAttributes(), this::visitChosenElementAttribute));
```

Comment:
What surprised me here is that we do two passes through the same method depending on whether a version is found and assigned to a field; that had me scratching my head a bit wondering if there's a different way. From the tests I can see it works, but for maintainability I'd like to briefly explore alternatives.

Reply:
I've made some changes that make the code more explicit about what we are doing and add more semantics and readability.
This way we are still doing two passes, one to validate the version and another one to maybe change the namespace attribute if it matches the oldValue, but all the boolean expressions are in methods that explain what they are about.
If you don't like the stream API, you can change the matchesVersion with:
```
if (versionMatcher != null) {
    for (Xml.Attribute attribute : tag.getAttributes()) {
        if (isVersionAttribute(attribute) && isVersionMatch(attribute)) {
            return true;
        }
    }
    return false;
}
return true;
```